covid_xprize/examples/prescriptors/neat/prescribe.py

```diff
@@ -155,41 +166,52 @@ def prescribe(start_date_str: str,
                 # Map prescription to integer outputs
                 prescribed_ips = (prescribed_ips * ip_max_values_arr).round()
 
-                # Add it to prescription dictionary
+                # Add it to prescription dictionary for the full ACTION_DURATION
                 country_name, region_name = geo.split('__')
                 if region_name == 'nan':
                     region_name = np.nan
-                df_dict['CountryName'].append(country_name)
-                df_dict['RegionName'].append(region_name)
-                df_dict['Date'].append(date_str)
-                for ip_col, prescribed_ip in zip(IP_COLS, prescribed_ips):
-                    df_dict[ip_col].append(prescribed_ip)
+                for date in pd.date_range(action_start_date, periods=ACTION_DURATION):
+                    if date > end_date:
+                        break
+                    date_str = date.strftime("%Y-%m-%d")
+                    df_dict['CountryName'].append(country_name)
+                    df_dict['RegionName'].append(region_name)
+                    df_dict['Date'].append(date_str)
+                    for ip_col, prescribed_ip in zip(IP_COLS, prescribed_ips):
+                        df_dict[ip_col].append(prescribed_ip)
 
             # Create dataframe from prescriptions
             pres_df = pd.DataFrame(df_dict)
 
             # Make prediction given prescription for all countries
             pred_df = get_predictions(start_date_str, date_str, pres_df)
 
-            # Update past data with new day of prescriptions and predictions
+            # Update past data with new days of prescriptions and predictions
             pres_df = add_geo_id(pres_df)
             pred_df = add_geo_id(pred_df)
```

Comment:
Same as above, these lines are indented so they run for the entire ACTION_DURATION.